### PR TITLE
Relevant order of the suggested tags #6582

### PR DESCRIPTION
**Pull request by atulonruby**

### CODE REVIEW

**Comment by atulonruby** [comment]
https://github.com/diaspora/diaspora/issues/6582

**Comment by jhass** [comment]
Is there a way to write a Jasmine spec for this?

File: app/assets/javascripts/helpers/js_object_helper.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+var sort_array_of_obj = function(data, property_name){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/helpers/js_object_helper.js]: Identifier 'sort_array_of_obj' is not in camel case.

Comment (diaspora-code-review) [inline app/assets/javascripts/helpers/js_object_helper.js]: Identifier 'property_name' is not in camel case.

File: app/assets/javascripts/helpers/js_object_helper.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+  return data.sort( function( obj1, obj2 ){
+    return obj1[property_name].localeCompare(obj2[property_name]);
```
Comment (diaspora-code-review) [inline app/assets/javascripts/helpers/js_object_helper.js]: Identifier 'property_name' is not in camel case.

File: app/assets/javascripts/helpers/js_object_helper.js (hunk 1 above)
Comment (diaspora-code-review) [inline app/assets/javascripts/helpers/js_object_helper.js]: 'sort_array_of_obj' is defined but never used.

File: spec/javascripts/js_object_helper-spec.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+describe("SortArrayofObjects", function() {
+  it('sort object by property length', function(){
```
Comment (diaspora-code-review) [inline spec/javascripts/js_object_helper-spec.js]: Strings must use doublequote.

File: spec/javascripts/js_object_helper-spec.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    var obj = [];
+    obj.push({name: "moviesnamelist"});
+    obj.push({name: "moviesnews"});
+    obj.push({name: "moviesn"});
+    expect(sort_array_of_obj(obj,"name")[0]).toEqual({name: "moviesn"});
```
Comment (diaspora-code-review) [inline spec/javascripts/js_object_helper-spec.js]: Identifier 'sort_array_of_obj' is not in camel case.

Comment (diaspora-code-review) [inline spec/javascripts/js_object_helper-spec.js]: 'sort_array_of_obj' is not defined.

File: vendor/assets/javascripts/jquery.autoSuggest.custom.js [hunk 5]
```diff
@@ -46,7 +46,7 @@
 		  	selectionRemoved: function(elem){ elem.remove(); },
 		  	formatList: false, //callback function
 		  	beforeRetrieve: function(string){ return string; },
-		  	retrieveComplete: function(data){ return data; },
+		  	retrieveComplete: function(data){ return sort_array_of_obj(data,"name"); },
```
Comment (diaspora-code-review) [inline vendor/assets/javascripts/jquery.autoSuggest.custom.js]: Identifier 'sort_array_of_obj' is not in camel case.

File: app/assets/javascripts/helpers/js_object_helper.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+var sortArrayOfObj = function(data, propertyName){
```
Comment (diaspora-code-review) [inline app/assets/javascripts/helpers/js_object_helper.js]: 'sortArrayOfObj' is defined but never used.

File: spec/javascripts/js_object_helper-spec.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+describe("sortArrayofObjects", function() {
+  it("sort object by property length", function(){
+    var obj = [];
+    obj.push({name: "moviesnamelist"});
+    obj.push({name: "moviesnews"});
+    obj.push({name: "moviesn"});
+    expect(sortArrayOfObj(obj,"name")[0]).toEqual( {name: "moviesn"} );
```
Comment (diaspora-code-review) [inline spec/javascripts/js_object_helper-spec.js]: 'sortArrayOfObj' is not defined.

File: vendor/assets/javascripts/jquery.autoSuggest.custom.js [hunk 8]
```diff
@@ -46,7 +46,7 @@
 		  	selectionRemoved: function(elem){ elem.remove(); },
 		  	formatList: false, //callback function
 		  	beforeRetrieve: function(string){ return string; },
-		  	retrieveComplete: function(data){ return data; },
+		  	retrieveComplete: function(data){ return sortArrayOfObj(data, "name"); },
```
Comment (jhass) [inline vendor/assets/javascripts/jquery.autoSuggest.custom.js]: This seems like a unnecessary modification, please overwrite it on the library calls instead.

File: app/assets/javascripts/helpers/js_object_helper.js [hunk 9]
```diff
@@ -0,0 +1,5 @@
...
+  return data.sort( function( obj1, obj2 ){
+    return obj1[propertyName].localeCompare( obj2[propertyName] );
+  });
+};
```
Comment (jhass) [inline app/assets/javascripts/helpers/js_object_helper.js]: Not sure this is the best place, @svbergerem @denschub thoughts?

Comment (AugierLe42e) [inline app/assets/javascripts/helpers/js_object_helper.js]: I think [`_.sortBy`](http://underscorejs.org/#sortBy) can be used instead.

Comment (denschub) [inline app/assets/javascripts/helpers/js_object_helper.js]: Yes, please.

**Comment by jhass** [comment]
Using a library function doesn't mean we shouldn't assert that it is used. A spec ensuring the elements are written in the right order into the HTML would still be great.

**Comment by denschub** [comment]
Ping.

**Comment by denschub** [comment]
Superseded and solved by #6734.

**Pull request closed without merging**